src/map/status.c

```diff
@@ -13397,25 +13397,127 @@ static bool status_readdb_sizefix(char *fields[], int columns, int current)
 	return true;
 }
 
-static bool status_readdb_scconfig(char *fields[], int columns, int current)
+static bool status_read_scdb_libconfig(void)
 {
-	int val = 0;
-	char* type = fields[0];
+	struct config_t status_conf;
+	const char *config_filename = "db/sc_config.conf"; // FIXME hardcoded name
 
-	nullpo_retr(false, fields);
-	if( !script->get_constant(type, &val) ){
-		ShowWarning("status_readdb_sc_conf: Invalid status type %s specified.\n", type);
+	if (libconfig->load_file(&status_conf, config_filename) == CONFIG_FALSE) {
+		ShowError("status_read_scdb_libconfig: can't read %s\n", config_filename);
 		return false;
 	}
 
-	status->dbs->sc_conf[val] = (int)strtol(fields[1], NULL, 0);
-	if (status->dbs->sc_conf[val] & SC_VISIBLE)
-	{
-		status->dbs->DisplayType[val] = true;
+	int i = 0;
+	int count = 0;
+	struct config_setting_t *it = NULL;
+
+	while ((it = libconfig->setting_get_elem(status_conf.root, i++)) != NULL) {
+		if (status->read_scdb_libconfig_sub(it, i - 1, config_filename))
+			++count;
 	}
 
+	libconfig->destroy(&status_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", count, config_filename);
+	return true;
+}
+
+static bool status_read_scdb_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+
+	int i32;
+	const char *name = config_setting_name(it);
+
+	if (!script->get_constant(name, &i32) || i32 < SC_NONE || i32 >= SC_MAX) {
+		ShowWarning("status_read_scdb_libconfig_sub: Invalid status type (%s) in \"%s\" entry #%d, skipping.\n", name, source, idx);
+		return false;
+	}
+
+	struct config_setting_t *fg = libconfig->setting_get_member(it, "Flags");
+	status->read_scdb_libconfig_sub_flag(fg, i32, source);
+
+	return true;
+}
+
+static bool status_read_scdb_libconfig_sub_flag(struct config_setting_t *it, int type, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+	Assert_retr(false, type >= SC_NONE && type < SC_MAX);
+
+	int i = 0;
+	struct config_setting_t *t = NULL;
+	while ((t = libconfig->setting_get_elem(it, i++)) != NULL) {
+		const char *flag = config_setting_name(t);
+		bool on = libconfig->setting_get_bool_real(t);
+
+		if (strcmpi(flag, "NoDeathReset") == 0) {
```

Comment:
you can change this into an array of structure holding the config name and bitmask value, then loop for each value found to match it with one of those array members, this will simplify the code and make it simpler to add more values in the future, also please do a check twice one with strcmp and one with strcmpi, if strcmpi success and strcmp fail give a warning, case insensitivity shall not tolerated.